python/src/py/objfloat.c

```diff
@@ -201,6 +201,24 @@ mp_obj_t mp_obj_new_float(mp_float_t value) {
     return MP_OBJ_FROM_PTR(o);
 }
 
+mp_obj_t mp_obj_new_float_via_str(const char* value) {
```

Comment:
Why don't you create a mp_float_t directly instead of allocing extra mp_obj_t on the Python heap?